test/test_polygon.py

```diff
@@ -91,6 +91,43 @@ def test_construction_frompolygon(self):
         self.assertEqual(po_2.vertices, [p1, p2, p3, p4])
         self.assertEqual(po_2.vertices, po.vertices)
 
+    def test_subscript(self):
+        """Checks whether reassigning a vertex works correctly"""
+        po = Polygon([p1, p2, p3, p4])
+
+        po[0] = (70.0, 80.0)
+        self.assertEqual(po[0], (70.0, 80.0))
+
+        po[1] = [45.0, 38.0]
+        self.assertEqual(po[1], (45.0, 38.0))
```

Comment:
I'd like to see:
- [ ] Every valid index tested, both positive and negative ( `0->3` and `-1 -> -3` )
- [ ] Invalid positive index tested, both super close to the list size and super far
- [ ] Invalid negative index ( in this case i.e. -4 and -100)
- [ ] Assigning a vertex to itself won't change the vertex